WordPress/openverse
# Set rate-limit headers on each response

## Fixes




Fixes #755

## Description




This PR
- updates audio views to use the same throttling as corresponding image views
- drops an unused throttle class
- documents the non-recoverable nature of the API keys
- adds `X-RateLimit-` headers to the responses to keep tabs on API usage
- documents the presence of these headers

## Testing Instructions


0. Check out the PR and start the services.
1. Read the documentation at `http://localhost:50280/v1/`.
2. Set `DISABLE_GLOBAL_THROTTLING` env var to `false` in `env.docker`.
3. Make unauthenticated requests and see your `X-RateLimit-Available` value go down to zero.
4. Start getting 429 responses.
5. Make authenticated requests and see your much higher `X-RateLimit-Available` value.

## Checklist



      `Update index.md`).
      a parent feature branch.
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin



<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
